### PR TITLE
Hide long description of settings 

### PR DESCRIPTION
**Pull request by twigleingrid**
Show only the first line of the description of a setting. 
This makes all the settings the same size, giving a better overview. 
Display or hide the description according to the button click.

Show more:
qute-show-more

Show less:
show-less

### CODE REVIEW

**Review by toofar — COMMENTED** [review]
Can you talk more about what use cases you've considered that this change supports? I ask because I'm not completely sold on everything being the same size being more useful than have all the text available. Although I agree it looks better with all the rows the same size.

The motivation you've given is to "give a better overview". But I'm not sure the main use case of this page is to get an overview of settings. I assume people would be looking for a particular setting by either name, value or keyword. Maybe I am wrong but given the size of the page I would hope they were doing that by searching, not scrolling. So given that, this means that the set of searchable keywords on the page is now less, since the collapsed parts don't seem to show up when searching. Additionally, once you have found the setting you are looking for surely it is good to have all the instructions right there without having to click an extra button.

I don't work much with UIs though, nor do I use the settings page all that much. So I would be happy to hear other people's opinions.

**Comment by twigleingrid** [comment]
Thank you for your review!
Giving the fact that the descriptions are of different sizes, it felt more chaotic for me. I thought the short descriptions, the first line, might give an overview and the user can decide if they need more information.  
I also think someone would rather search, but I figured the main keywords would appear in the setting name or in the first line of the description. Also sometimes some scrolling or scheming through the settings' names is still necessary, so the short version felt more pleasant to me.
But I am interested in other opinions as well!

**Comment by mschilli87** [comment]
IMHO if there is a setting that can only be found by a keyword 'hidden' in the long description, we should treat this as an issue with the short one not being clear enough. I like the new layout and would expect to find what I need. The extra click to get the details is a cost though, I agree. Not sure how to weigh this best. My feeling is this will require an executive decision (probably by @The-Compiler) in the end and we'll just go with it from there.

**Comment by The-Compiler** [comment]
I don't really have a strong opinion on this, but I lean towards agreeing with @twigleingrid and @mschilli87.

As for the search, we might want to provide a more clever search for the page anyways - filtering all the displayed settings based on the searched term, and maybe even prioritizing the matches based on their relevance (matches on setting name > matches in first line of description > matches in full description).

It's even imaginable to special case `/` in qutebrowser to use that on-page search then - we might want to do something similar for e.g. PDFs displayed in the browser too, so it wouldn't be the only place something like that might be useful.

File: qutebrowser/html/settings.html [hunk 1]
```diff
@@ -10,6 +10,22 @@
   xhr.open("GET", url);
   xhr.send();
 }
+
+let displayDesc = (index) => {
+  let idDesc = `long-desc-${index}`
+  let longDesc = document.getElementById(idDesc)
+  let idButton = `show-more-${index}`
+  let button = document.getElementById(idButton)
+  let display = longDesc.style.display
```
Comment (The-Compiler) [inline qutebrowser/html/settings.html:19]: Minor thing, but I suppose those could be using `const` instead of `let`?

**Review by The-Compiler — CHANGES_REQUESTED** [review]
Something small I noticed while looking at the code.

I wonder though, wouldn't it be more appropriate to use [`<details>` elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) for this? 

This would mean less re-inventing stuff in JS, and at least in Chromium, searching for the part inside the `<details>` also opens the element. For some reason this doesn't happen with qutebrowser/QtWebEngine though, but it still seems like a potentially simpler approach (at least if it's possible to style appropriately in CSS).

**Comment by twigleingrid** [comment]
Thank you for reviewing, @The-Compiler! That's a nice idea to use the `details` tag!

**Review by toofar — APPROVED** [review]
Looks okay to me. The cursor does what I expect now which is the main thing I cared about!

Although if we want to quibble on making things the same size; I think the original Show less/more thing done in JS looked better because it was 1) inline when collapsed and 2) didn't interrupt the text when expanded. Those could maaaybe be addressed by making the first line of the setting text the summary, but that might make it less obvious that you should click on them.

**Review by toofar — APPROVED** [review]
That looks really good!

Edit: I don't think the test failures are related, you can ignore them.

https://user-images.githubusercontent.com/7419144/158114571-8a6c489f-4664-44f5-83ec-eb7173f87d6c.mp4

**Comment by twigleingrid** [comment]
Thank you @toofar for adding the little video! I forgot to include screenshots, will remember for next time :)

**Comment by toofar** [comment]
Thanks @twigleingrid, it was good seeing the iterative improvement!

**Pull request merged**